plugins/cpp_metrics/service/src/cppmetricsservice.cpp

```diff
@@ -81,6 +96,111 @@ double CppMetricsServiceHandler::getSingleCppMetricForAstNode(
   });
 }
 
+void CppMetricsServiceHandler::getCppMetricsForModule(
+  std::vector<CppMetricsModule>& _return,
+  const core::FileId& fileId_)
+{
+  CppMetricsModule metric;
+
+  _transaction([&, this](){
+    typedef odb::query<model::CppFileMetrics> CppModuleMetricsQuery;
+
+    auto moduleMetrics = _db->query<model::CppFileMetrics>(
+      CppModuleMetricsQuery::file == std::stoull(fileId_));
+
+    for (const auto& moduleMetric : moduleMetrics)
+    {
+      metric.type = static_cast<CppModuleMetricsType::type>(moduleMetric.type);
+      metric.value = moduleMetric.value;
+      _return.push_back(metric);
+    }
+  });
+}
+
+void CppMetricsServiceHandler::getCppAstNodeMetricsForPath(
+  std::vector<CppAllMetricsAstNode>& _return,
+  const std::string& path_)
+{
+  _transaction([&, this]()
+  {
+    typedef odb::query<model::CppAstNodeMetrics> CppAstNodeMetricsQuery;
+    typedef odb::result<model::CppAstNodeMetrics> CppAstNodeMetricsResult;
+    typedef odb::query<model::CppAstNodeFilePath> CppAstNodeFilePathQuery;
+    typedef odb::result<model::CppAstNodeFilePath> CppAstNodeFilePathResult;
+
+    CppAstNodeFilePathResult nodes = _db->query<model::CppAstNodeFilePath>(
+      CppAstNodeFilePathQuery::LocFile::path.like(path_ + '%'));
+
+    if (nodes.empty())
+    {
+      core::InvalidInput ex;
+      ex.__set_msg("Invalid metric type for path: " + path_);
```

Comment:
Incorrect error message: it was not the metric type, which was invalid, but no metrics were found for the given path.

Also, maybe returning an empty list, instead of raising an exception would be more fitting here? I mean, a path could be valid, but maybe there are no metrics calculated for it?

Reply:
Fixed in eb36b1b2b60fbf6e7909cbba169bc16ba37719ff.